### PR TITLE
DH-1563 solved search term highliting and escaping string input

### PR DESCRIPTION
**Pull request by yozzo**
Solved search term highliting and escaping string input of potentially hazardous characters

### CODE REVIEW

File: config/nunjucks/filters.js [hunk 1]
```diff
@@ -102,7 +103,8 @@ const filters = {
       const cleanTerm = searchTerm.replace(/[.*+?^${}()|[\]\\]/g, '') // $& means the whole matched string
 
       const regEx = new RegExp(`(${cleanTerm})${shouldMatchFullWord ? '\\b' : ''}`, 'gi')
-      const result = string.replace(regEx, '<span class="u-highlight">$1</span>')
+      const cleanString = escape(string)
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: `escapedString` could be a more appropriate name?

File: config/nunjucks/filters.js [hunk 2]
```diff
@@ -102,7 +103,8 @@ const filters = {
       const cleanTerm = searchTerm.replace(/[.*+?^${}()|[\]\\]/g, '') // $& means the whole matched string
 
       const regEx = new RegExp(`(${cleanTerm})${shouldMatchFullWord ? '\\b' : ''}`, 'gi')
-      const result = string.replace(regEx, '<span class="u-highlight">$1</span>')
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: not part of this work but `string`, IMO is a terrible variable name!
I would go with something which at a look describes what it is, rather than its type. What is `string`? what does it contain?

`textValue`? or `searchResultText`? instead of `string`

File: config/nunjucks/filters.js [hunk 3]
```diff
@@ -102,7 +103,8 @@ const filters = {
       const cleanTerm = searchTerm.replace(/[.*+?^${}()|[\]\\]/g, '') // $& means the whole matched string
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: I wonder what is the comment here referring to here? should it be removed?

Comment (feedmypixel) [inline config/nunjucks/filters.js]: Can I suggest `cleanedSearchTerm` rather than `cleanTerm`

File: config/nunjucks/filters.js [hunk 4]
```diff
@@ -102,7 +103,8 @@ const filters = {
...
 
       const regEx = new RegExp(`(${cleanTerm})${shouldMatchFullWord ? '\\b' : ''}`, 'gi')
```
Comment (feedmypixel) [inline config/nunjucks/filters.js]: Don't you also need a word boundary flag `\b` at the start?

Comment (feedmypixel) [inline config/nunjucks/filters.js]: Could this could be `matchFullWord` rather than `shouldMatchFullWord`.

It also feels like this is quite complicated, can I suggest:

```
const searchPattern = matchFullWord 
  ? new RegExp(`\\b(${cleanedTerm})\\b`, 'gi')
  : new RegExp(`(${cleanedTerm})`, 'gi')

const escapedString = escape(string)
const resultWithHighlight = escapedString.replace(searchPattern, '<span class="u-highlight">$1</span>')
```

File: config/nunjucks/filters.js (hunk 1 above)
Comment (yozzo) [reply config/nunjucks/filters.js]: amended

File: config/nunjucks/filters.js (hunk 2 above)
Comment (yozzo) [reply config/nunjucks/filters.js]: amended

File: config/nunjucks/filters.js (hunk 3 above)
Comment (yozzo) [reply config/nunjucks/filters.js]: removed

File: config/nunjucks/filters.js (hunk 4 above)
Comment (yozzo) [reply config/nunjucks/filters.js]: amended

File: test/unit/config/nunjucks/filters.test.js [hunk 5]
```diff
@@ -93,6 +93,26 @@ describe('nunjucks filters', () => {
         expect(this.highlightedString.val).to.equal('Here is an <span class="u-highlight">exam</span>ple phrase')
       })
     })
+
```
Comment (feedmypixel) [inline test/unit/config/nunjucks/filters.test.js]: Feels like it would be worth adding in a test around the word boundary work?

```
it('should highlight only a word')
it('should highlight matched text')
```

Comment (yozzo) [reply test/unit/config/nunjucks/filters.test.js]: added
